### PR TITLE
[Accessibility] Add a label to the product pagination for the woocommerce pagination

### LINKED ISSUE #47071: [Enhancement]: Add Navigation Landmarks
### Describe the solution you'd like

An accessibility best practice (and sometimes requirement) is to [add navigation landmarks to a page](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA11).

Users who use screen reading software will often use tools to help them skip repetitive parts of a website like the header, navigation, sidebar, etc. To give these tools as much information as possible we should add ARIA landmarks to these sections.

🔧 Example tool: [Landmark Navigation via Keyboard or Pop-up](https://chromewebstore.google.com/detail/landmark-navigation-via-k/ddpokpbjopmeeiiolheejjpkonlkklgp) Chrome browser extension

Here's what this tool outputs in an empty WooCommerce store:
Screenshot 2024-04-30 at 3 26 02 PM

Here's what it looks like on a live Woo store with other plugins & themes active: 
navigation-landmarks

You can see in the live Woo store that we have 3 nav elements in the Main section. Two of those are from WooCommerce core. One is from a plugin (Gridlist toggle).

It would be helpful if WooCommerce core and extensions would label their landmarks to help screen reading software scan the page and give users the ability to skip right to the section they want.

My preferred solution: edit the Pagination template (`woocommerce/plugins/woocommerce/templates/loop/pagination.php`) to include an aria label.

``` html
<nav class="woocommerce-pagination" aria-label="Product Pagination">
```

---

Here's what this looks like if both WooCommerce & extensions had better labels on nav elements:

Screenshot 2024-04-30 at 3 32 58 PM

The labeled navigation elements are much more useful to people who use screen reading software.

### Describe alternatives you've considered

You can also use `aria-labelledby` but it seems less intuitive than `aria-label`. 🤷 

### Additional context

Additional WCAG reading: https://www.w3.org/WAI/WCAG21/Understanding/bypass-blocks.html

### PR DESCRIPTION
**Pull request by amesplant**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #47071 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Visit the Store and investigate the code to ensure that the production navigation has the appropriate `aria-label`
```<nav class="woocommerce-pagination" aria-label="Product Pagination">```
2. In addition, view the site in Safari and launch VoiceOver, bring up the Rotor and navigate to Landmarks where the navigation landmark should now be names as **Product Pagination**

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Comment by amesplant** [comment]
@dkotter sending your way for Code Review

**Review by dkotter — APPROVED** [review]
@ankitguptaindia @qasumitbagthariya This is ready for QA

**Comment by qasumitbagthariya** [comment]
### QA Update ✅ 
-------------

I have checked this PR in the `amesplant:fix/47071-navigation-landmarks` branch and changes working as expected.

1. Visit the Store and investigate the code to ensure that the production navigation has the appropriate aria-label
`<nav class="woocommerce-pagination" aria-label="Product Pagination">`  ✅ 

2. In addition, view the site in Safari and launch VoiceOver, bring up the Rotor and navigate to Landmarks where the navigation landmark should now be names as Product Pagination ✅ 

**Testing Environment** 

<Details>

- WordPress: 6.6
- Theme: Storefront 4.6.0
- WooCommerce - WooCommerce - Dev version with this fix specific branch.
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS Ventura 13.3 
- Git Branch: `amesplant:fix/47071-navigation-landmarks`

</Details>

image
image

**Comment by qasumitbagthariya** [comment]
@amesplant Please take this PR out of the draft.

**Comment by nielslange** [comment]
I'll remove myself from the review, given that this PR affects the product archive (store page), while I'm maintaining the Cart and Checkout blocks.

**Comment by amesplant** [comment]
Thanks @nielslange

**Comment by vikrampm1** [comment]
@woocommerce/rubik this PR is ready for your review.

**Review by naman03malhotra — APPROVED** [review]
This change looks, good thanks for working on it.

File: plugins/woocommerce/templates/loop/pagination.php [hunk 1]
```diff
@@ -12,7 +12,7 @@
  *
  * @see     https://woocommerce.com/document/template-structure/
  * @package WooCommerce\Templates
- * @version 3.3.1
+ * @version 9.2.0
```
Comment (dinhtungdu) [inline plugins/woocommerce/templates/loop/pagination.php]: Can you bump this to 9.3.0 as this PR won't get released in 9.2?

Comment (dinhtungdu) [reply plugins/woocommerce/templates/loop/pagination.php]: You can get the information about the current release in [Woo Developer Blog](https://developer.woocommerce.com). For example, [this post](https://developer.woocommerce.com/2024/07/26/woocommerce-9-2-pre-release-updates/#h-update-timeline) contains all information related to the 9.2 release.

Comment (amesplant) [reply plugins/woocommerce/templates/loop/pagination.php]: @dinhtungdu bumped to 9.3.0

**Review by dinhtungdu — APPROVED** [review]
Thanks for the update! This is LGTM now.

**Pull request merged**